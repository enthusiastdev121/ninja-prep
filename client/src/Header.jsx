import React from 'react'

function Header() {
	return <div> Ninja-Prep!! </div>
}

export default Header
