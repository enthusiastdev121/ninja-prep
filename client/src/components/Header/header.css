/* Wave image is not responsive */
.header-bg {
  background-image: url('/images/wave.png');
  background-size: cover;
  background-repeat: no-repeat;
  width: auto;
  height: 100%;
  background-position: center center;
}

.header-bg-mobile {
  background-image: url('/images/wave.png');
  background-size: cover;
  background-repeat: no-repeat;
  height: 75%;
  width: auto;
  background-position: center center;
}

.header-container {
  position: relative;
  top: 15%;
}

.header-container-mobile {
  position: relative;
  top: 12%;
}
.form-bg {
  background-color: #404053;
}
