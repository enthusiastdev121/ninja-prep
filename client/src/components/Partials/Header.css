.btn-flat {
    background-color: purple !important;
    color: white !important;
}

.btn-xxl {
    padding: 1rem 1.5rem !important;
    font-size: 1.5rem !important;
}
