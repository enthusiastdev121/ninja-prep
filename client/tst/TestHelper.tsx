import React from 'react'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
const history = createBrowserHistory()

interface DEVICE_WIDTHS {
    readonly DESKTOP: number
    readonly MOBILE: number
    readonly TABLET: number
}

//Widths in Pixels. Aligns with Media Query widths
export const DEVICE_TYPES: DEVICE_WIDTHS = {
    DESKTOP: 1000,
    MOBILE: 750,
    TABLET: 800
}

export const routeWrapper = (children: React.ReactChild) => {
    return <Router history={history}>{children}</Router>
}
