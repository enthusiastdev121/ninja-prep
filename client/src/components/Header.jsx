import React from 'react'
import '../css/header.css'

const Header = (props) => {
	return (
		<div class="header">
			<div class="header-content">
				<h1>A place to gain new coding skills</h1>
				<h2>
					NinjaPrep will help you build a strong and solid foundation on data
					structures and algorithms to prepare you for technical interviews.
				</h2>

				<p class="bold">Get started</p>
			</div>
		</div>
	)
}

export default Header
export { default as Header } from './Header'
