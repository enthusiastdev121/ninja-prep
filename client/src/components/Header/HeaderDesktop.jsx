import React from 'react'
import { Link } from 'react-router-dom'
import NavigateNextIcon from '@material-ui/icons/NavigateNext'
import { Button } from 'react-bootstrap'
import './header.css'


const HeaderDesktop = () => (
	<div className="py-5 text-center header-bg">
		<div className="py-3 px-5 ">
			<img
				className="w-auto h-20"
				src="/images/NinjaPrep-Logo.png"
				alt="logo"
			/>

			<h1 className="display-4 px-5 font-weight-bold">
				Enhance your skills here
			</h1>

			<span className="lead w-75 m-auto d-block">
				A place to help you build a strong foundation in data structures and
			</span>
			<span className="lead w-75 m-auto d-block">
				algorithms and prepare for technical interviews, all in one platform.
			</span>

			<Link to="/login">
				<Button
					variant="gradient"
					className=" mt-2 mt-5 px-4 font-weight-bold btn-lg"
				>
					Start Learning <NavigateNextIcon className="start-learning-icon" />
				</Button>
			</Link>
		</div>
	</div>
)

export default HeaderDesktop
export { default as HeaderDesktop } from './HeaderDesktop'
