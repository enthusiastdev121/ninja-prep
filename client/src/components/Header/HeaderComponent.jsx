import React from 'react'
import {HeaderDesktop} from './HeaderDesktop'
import {HeaderTablet} from './HeaderTablet'
import {HeaderMobile} from './HeaderMobile'
import {Desktop, Tablet, Mobile } from '../MediaQuery/DeviceSizes'


const HeaderComponent = () => (
	<div>
		<Desktop>
			<HeaderDesktop />
		</Desktop>
		<Tablet>
			<HeaderTablet />
		</Tablet>
		<Mobile>
			<HeaderMobile />
		</Mobile>
	</div>
)

export default HeaderComponent
export { default as HeaderComponent } from './HeaderComponent'
