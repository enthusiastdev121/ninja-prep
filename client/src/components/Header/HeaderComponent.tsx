/*
Top landing page content for when user visits ninjaprep.io
*/
import React from 'react'
import HeaderDesktop from './HeaderDesktop'
import HeaderTablet from './HeaderTablet'
import HeaderMobile from './HeaderMobile'
import { Desktop, Tablet, Mobile } from '../MediaQuery/DeviceSizes'

const HeaderComponent = () => (
    <div>
        <Desktop>
            <HeaderDesktop />
        </Desktop>
        <Tablet>
            <HeaderTablet />
        </Tablet>
        <Mobile>
            <HeaderMobile />
        </Mobile>
    </div>
)

export default HeaderComponent
