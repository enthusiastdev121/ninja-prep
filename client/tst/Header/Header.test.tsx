import React from 'react'
import Header from '../../src/components/Header/HeaderComponent'
import { shallow, mount } from 'enzyme'
import { Context as ResponsiveContext } from 'react-responsive'
import { DEVICE_TYPES } from '../TestHelper'
import toJson from 'enzyme-to-json'

test.each(Object.entries(DEVICE_TYPES))('renders successfully on %p', (deviceType: string, deviceWidth: number) => {
    const wrappingComponent = ResponsiveContext.Provider
    const wrappingComponentProps = { value: { width: deviceWidth } }
    const mountProps = { wrappingComponent, wrappingComponentProps }

    const wrapper = shallow(<Header />, mountProps)
    expect(wrapper).toBeDefined
})

test.each(Object.entries(DEVICE_TYPES))('matches snapshot on %p', (deviceType: string, deviceWidth: number) => {
    const wrappingComponent = ResponsiveContext.Provider
    const wrappingComponentProps = { value: { width: deviceWidth } }
    const mountProps = { wrappingComponent, wrappingComponentProps }

    const wrapper = mount(<Header />, mountProps)
    expect(toJson(wrapper)).toMatchSnapshot()
})
