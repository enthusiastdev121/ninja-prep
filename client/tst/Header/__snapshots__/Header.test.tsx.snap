// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot on "DESKTOP" 1`] = `
<HeaderComponent>
  <div>
    <Desktop>
      <HeaderDesktop>
        <div
          className="header-bg text-light px-5"
        >
          <Container
            className="header-container"
            fluid={false}
          >
            <div
              className="header-container container"
            >
              <Row
                noGutters={false}
              >
                <div
                  className="row"
                >
                  <Col
                    className="pr-5 pl-0 mx-5 my-auto"
                  >
                    <div
                      className="pr-5 pl-0 mx-5 my-auto col"
                    >
                      <div>
                        <h3
                          className="font-weight-bold pb-2"
                        >
                          Learn new coding skills here
                        </h3>
                        <p
                          className="lead"
                        >
                          A place to help you build a strong foundation in data structures and
                           
                          algorithms and prepare for technical interviews, all in one platform.
                        </p>
                      </div>
                    </div>
                  </Col>
                  <Col
                    className="pl-5 pr-0 mx-5"
                  >
                    <div
                      className="pl-5 pr-0 mx-5 col"
                    >
                      <div
                        className="form-bg rounded"
                      >
                        <div
                          className="pt-4 pb-2 px-5"
                        >
                          <h4
                            className="font-weight-bold pb-3 mx-auto text-light"
                          >
                            Get Started for Free
                          </h4>
                          <SocialMediaButtons
                            cssClass="mx-auto fluid ui mb-4 py-3"
                          >
                            <div>
                              <div
                                className="lead"
                              >
                                <a
                                  className="mx-auto fluid ui mb-4 py-3 red google button"
                                  href="/api/auth/google"
                                >
                                  <i
                                    className="google icon"
                                  />
                                  Google Login
                                </a>
                              </div>
                              <div
                                className="lead"
                              >
                                <a
                                  className="mx-auto fluid ui mb-4 py-3 facebook button"
                                  href="/api/auth/facebook"
                                >
                                  <i
                                    className="facebook icon"
                                  />
                                  Facebook Login
                                </a>
                              </div>
                              <div
                                className="lead"
                              >
                                <a
                                  className="mx-auto fluid ui mb-4 py-3 black button"
                                  href="/api/auth/github"
                                >
                                  <i
                                    className="github icon"
                                  />
                                  Github Login
                                </a>
                              </div>
                            </div>
                          </SocialMediaButtons>
                        </div>
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </Container>
        </div>
      </HeaderDesktop>
    </Desktop>
    <Tablet />
    <Mobile />
  </div>
</HeaderComponent>
`;

exports[`matches snapshot on "MOBILE" 1`] = `
<HeaderComponent>
  <div>
    <Desktop />
    <Tablet />
    <Mobile>
      <HeaderMobile>
        <div
          className="header-bg-mobile"
        >
          <div
            className="mx-5 header-container-mobile"
          >
            <Row
              className=""
              noGutters={false}
            >
              <div
                className="row"
              >
                <div
                  className="mx-auto text-light"
                >
                  <div
                    className="pb-2"
                  >
                    <h3
                      className="font-weight-bold pb-3 mx-auto text-light"
                    >
                      Start learning for free with NinjaPrep
                    </h3>
                    <div>
                      <SocialMediaButtons
                        cssClass="mx-auto fluid ui mb-4 py-3 "
                      >
                        <div>
                          <div
                            className="lead"
                          >
                            <a
                              className="mx-auto fluid ui mb-4 py-3  red google button"
                              href="/api/auth/google"
                            >
                              <i
                                className="google icon"
                              />
                              Google Login
                            </a>
                          </div>
                          <div
                            className="lead"
                          >
                            <a
                              className="mx-auto fluid ui mb-4 py-3  facebook button"
                              href="/api/auth/facebook"
                            >
                              <i
                                className="facebook icon"
                              />
                              Facebook Login
                            </a>
                          </div>
                          <div
                            className="lead"
                          >
                            <a
                              className="mx-auto fluid ui mb-4 py-3  black button"
                              href="/api/auth/github"
                            >
                              <i
                                className="github icon"
                              />
                              Github Login
                            </a>
                          </div>
                        </div>
                      </SocialMediaButtons>
                    </div>
                  </div>
                </div>
              </div>
            </Row>
          </div>
        </div>
      </HeaderMobile>
    </Mobile>
  </div>
</HeaderComponent>
`;

exports[`matches snapshot on "TABLET" 1`] = `
<HeaderComponent>
  <div>
    <Desktop />
    <Tablet>
      <HeaderTablet>
        <div
          className="header-bg text-light"
        >
          <Container
            className="header-container px-5"
            fluid={false}
          >
            <div
              className="header-container px-5 container"
            >
              <Row
                noGutters={false}
              >
                <div
                  className="row"
                >
                  <Col
                    className="my-auto pr-4 pl-0"
                  >
                    <div
                      className="my-auto pr-4 pl-0 col"
                    >
                      <div>
                        <h3
                          className="font-weight-bold pb-2"
                        >
                          Learn new coding skills here
                        </h3>
                        <p
                          className="lead"
                        >
                          A place to help you build a strong foundation in data structures and
                           
                          algorithms and prepare for technical interviews, all in one platform.
                        </p>
                      </div>
                    </div>
                  </Col>
                  <Col
                    className="pl-3 pr-0"
                  >
                    <div
                      className="pl-3 pr-0 col"
                    >
                      <div
                        className="form-bg rounded"
                      >
                        <div
                          className="pt-4 pb-2 px-5"
                        >
                          <h4
                            className="font-weight-bold pb-3 mx-auto text-light"
                          >
                            Get Started for Free
                          </h4>
                          <div>
                            <SocialMediaButtons
                              cssClass="mx-auto fluid ui mb-4 py-3"
                            >
                              <div>
                                <div
                                  className="lead"
                                >
                                  <a
                                    className="mx-auto fluid ui mb-4 py-3 red google button"
                                    href="/api/auth/google"
                                  >
                                    <i
                                      className="google icon"
                                    />
                                    Google Login
                                  </a>
                                </div>
                                <div
                                  className="lead"
                                >
                                  <a
                                    className="mx-auto fluid ui mb-4 py-3 facebook button"
                                    href="/api/auth/facebook"
                                  >
                                    <i
                                      className="facebook icon"
                                    />
                                    Facebook Login
                                  </a>
                                </div>
                                <div
                                  className="lead"
                                >
                                  <a
                                    className="mx-auto fluid ui mb-4 py-3 black button"
                                    href="/api/auth/github"
                                  >
                                    <i
                                      className="github icon"
                                    />
                                    Github Login
                                  </a>
                                </div>
                              </div>
                            </SocialMediaButtons>
                          </div>
                        </div>
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </Container>
        </div>
      </HeaderTablet>
    </Tablet>
    <Mobile>
      <HeaderMobile>
        <div
          className="header-bg-mobile"
        >
          <div
            className="mx-5 header-container-mobile"
          >
            <Row
              className=""
              noGutters={false}
            >
              <div
                className="row"
              >
                <div
                  className="mx-auto text-light"
                >
                  <div
                    className="pb-2"
                  >
                    <h3
                      className="font-weight-bold pb-3 mx-auto text-light"
                    >
                      Start learning for free with NinjaPrep
                    </h3>
                    <div>
                      <SocialMediaButtons
                        cssClass="mx-auto fluid ui mb-4 py-3 "
                      >
                        <div>
                          <div
                            className="lead"
                          >
                            <a
                              className="mx-auto fluid ui mb-4 py-3  red google button"
                              href="/api/auth/google"
                            >
                              <i
                                className="google icon"
                              />
                              Google Login
                            </a>
                          </div>
                          <div
                            className="lead"
                          >
                            <a
                              className="mx-auto fluid ui mb-4 py-3  facebook button"
                              href="/api/auth/facebook"
                            >
                              <i
                                className="facebook icon"
                              />
                              Facebook Login
                            </a>
                          </div>
                          <div
                            className="lead"
                          >
                            <a
                              className="mx-auto fluid ui mb-4 py-3  black button"
                              href="/api/auth/github"
                            >
                              <i
                                className="github icon"
                              />
                              Github Login
                            </a>
                          </div>
                        </div>
                      </SocialMediaButtons>
                    </div>
                  </div>
                </div>
              </div>
            </Row>
          </div>
        </div>
      </HeaderMobile>
    </Mobile>
  </div>
</HeaderComponent>
`;
